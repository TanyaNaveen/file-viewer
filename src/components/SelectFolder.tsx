import { useEffect } from "react"
import React from "react";
import Dropdown from 'react-dropdown';
import styles from "./styles/SelectFolder.module.css"; 
import "./styles/Dropdown.css";


const SelectFolder = (
    {handleGo}: {handleGo: (folder: string) => void}
) => {

    // set options to display in the dropdown
    const [options, setOptions] = React.useState<string[]>([]);
    // track which option the user has currently chosen
    const [currFolder, setCurrFolder] = React.useState("");;
    
    useEffect(() => {
        // fetch list of all folders in public directory
        const fetchFolder = async () => {
            const response = await fetch(`/./data/directory.json`)
              .then((response) => response.json())
              .then((v) => v)
              .catch((err) => console.log(err));
            return response;
        };
          
        // create array of folders to display in dropdown
        fetchFolder().then((dirData: { folder: string }[]) => {
            const options: string[] = []
            for (let i = 0; i < dirData.length; i++) {
                options.push(dirData[i].folder)
            }
            setOptions(options)
        });

    }, []);

    // keep track of which option the user chooses
    const handleChange = (evt: any) => {
        if (evt.label !== undefined) {
            setCurrFolder(evt.label)
        }
    }
    
    return (
        <div className={styles.container}>
            <div className={styles.dropdown} data-testid="dropdown">
                <Dropdown  options={options} value={currFolder} onChange={handleChange} placeholder="Select an option" />
            </div>
            <button className={styles.goButton} data-testid="button" onClick={() => handleGo(currFolder)}>Go</button>

        </div>
    )
}

export default SelectFolder