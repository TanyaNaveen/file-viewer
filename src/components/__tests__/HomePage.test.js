import { render, screen, cleanup, fireEvent, getAllByRole } from "@testing-library/react";
import SelectFolder from "../SelectFolder";

afterEach(() => {
    cleanup(); // Resets DOM after each test suite
})

describe('Home Page', () => {

    it('renders Home page correctly', () => {
        const { container } = render(<SelectFolder />);
        expect(container).toMatchSnapshot();
    });

    // want to check that clicking the go button actually goes to next page
    // only when an option has been selected
    // otherwise it should alert the user    

    test ('go button renders', () => {
        render(<SelectFolder />)
        expect(screen.getByRole('button', { name: 'Go'})).toBeInTheDocument();
    })

    test ('go button: does nothing when no option selected', () => {
        const { container } = render(<SelectFolder handleGo={() => 2} />); // pass in dummy callback
        const button = screen.getByTestId("button");

        fireEvent.click(button);
        expect(container).toMatchSnapshot(); // since no option has been selected, page should not change
        
        jest.spyOn(window, 'alert').mockImplementation(() => {});

    
    })

    it('dropdown renders', () => {
        render(<SelectFolder />)
        const dropdown = screen.getByTestId("dropdown");
        const display = dropdown.children[0];
        expect(display.textContent).toBe("Select an option");
        fireEvent.click(dropdown);
        // screen.getByRole('option');
        // const dropdownOptions = getAllByRole(dropdown, 'option', {hidden: true});

    });
})