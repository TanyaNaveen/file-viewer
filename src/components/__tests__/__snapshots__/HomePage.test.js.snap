// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page go button: does nothing when no option selected 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="dropdown"
      data-testid="dropdown"
    >
      <div
        class="Dropdown-root"
      >
        <div
          aria-haspopup="listbox"
          class="Dropdown-control"
        >
          <div
            class="Dropdown-placeholder"
          >
            Select an option
          </div>
          <div
            class="Dropdown-arrow-wrapper"
          >
            <span
              class="Dropdown-arrow"
            />
          </div>
        </div>
      </div>
    </div>
    <button
      class="goButton"
      data-testid="button"
    >
      Go
    </button>
  </div>
</div>
`;

exports[`Home Page renders Home page correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="dropdown"
      data-testid="dropdown"
    >
      <div
        class="Dropdown-root"
      >
        <div
          aria-haspopup="listbox"
          class="Dropdown-control"
        >
          <div
            class="Dropdown-placeholder"
          >
            Select an option
          </div>
          <div
            class="Dropdown-arrow-wrapper"
          >
            <span
              class="Dropdown-arrow"
            />
          </div>
        </div>
      </div>
    </div>
    <button
      class="goButton"
      data-testid="button"
    >
      Go
    </button>
  </div>
</div>
`;
